src/test/java/org/sonarsource/scanner/cli/MainTest.java

```diff
@@ -95,33 +96,36 @@ public void should_stop_on_error() {
   }
 
   @Test
-  public void show_error_stacktrace() {
-    Exception e = show_error(false, true);
-    verify(logs).error("Error during SonarQube Scanner execution", e);
+  public void show_error() {
+    show_error(false, false);
+    verify(logs).error("Error during SonarQube Scanner execution");
     verify(logs).error("Re-run SonarQube Scanner using the -X switch to enable full debug logging.");
   }
-
+  
   @Test
-  public void show_error_debug() {
-    Exception e = show_error(true, false);
-
-    verify(logs).error("Error during SonarQube Scanner execution", e);
-    verify(logs, never()).error("Re-run SonarQube Scanner using the -X switch to enable full debug logging.");
+  public void show_error_MessageException() {
+    show_error(false, true);
+    verify(logs).error("Error during SonarQube Scanner execution");
+    verify(logs).error("Re-run SonarQube Scanner using the -X switch to enable full debug logging.");
   }
 
   @Test
-  public void show_error_debug_stack() {
-    Exception e = show_error(true, true);
+  public void show_error_debug() {
+    Exception e = show_error(true, false);
 
     verify(logs).error("Error during SonarQube Scanner execution", e);
     verify(logs, never()).error("Re-run SonarQube Scanner using the -X switch to enable full debug logging.");
   }
 
-  private Exception show_error(boolean debugEnabled, boolean stackTraceEnabled) {
-    Exception e = new NullPointerException("NPE");
-    e = new IllegalStateException("Error", e);
+  private Exception show_error(boolean debugEnabled, boolean messageException) {
```

Comment:
This method *smells*... The name suggests to "show" something, at the same time it also creates and returns an `Exception`. Also, `snake_case` is fine in test methods, but this is not a test method.